### ISSUE TITLE
Selecting Edit or Pan tool aborts path drawing

### ISSUE THREAD

**Issue by lpechacek** [OpenOrienteering/mapper#1175]
### Steps to reproduce
1. Select Path drawing tool and start drawing a line
2. Choose Edit or Pan tool

### Actual behaviour
The freshly drawn line is lost.

### Expected behaviour
Realizing that I'm not the only user I'm listing multiple options:
1) Edit and Pan tools are inactive like the other tools
2) The line gets finished instead of aborted
3) Mapper warns about potential loss of data with OK/Cancel dialog

My preference is towards finishing the line (option 2) and I'd be happy to hear both supporting and contradicting opinions on the solution.

### Configuration
Mapper Version: 0.8.3
Operating System: All

**Comment by dg0yt** [comment]
Understood. 

IMO a dialog (3) is a no-go. With regard to Android, finishing the line (2) might not be what you want in touch mode when clicking on the Pan tool?

**Comment by ollesmaps** [comment]
I have an option 4. The line remains "half-finished" and you can continue drawing after panning. 
That would greatly enhance the fluency of drawing as mentioned in #988.

**Comment by lpechacek** [comment]
> With regard to Android, finishing the line (2) might not be what you want in touch mode when clicking on the Pan tool?

Absolutely. Thanks for the reminder. As Olles pointed out there is open issue to allow panning while drawing. I recalled that one yesterday but kept unconscious of it while writing the issue comment. I'll update the above options accordingly.

**Comment by lpechacek** [comment]
The above pull request implements part one of option two. I'll look into interaction with Pan tool in the corresponding issue.

**Comment by dg0yt** [comment]
Dekuji moc, Libor. Indeed, I will try to do v0.8.4 soon.

(For v0.9.0, I'm working on getting Qt 5.12 ready, and I'm on good track. Probably in-time for 5.12.1 which fixes some issues for us.)

**Comment by lpechacek** [comment]
Gern getan!
